python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -593,33 +593,17 @@ def info_string(self, nodes):
 
         return "{} nodes{}".format(len(nodes), suffix)
 
-    def request_resources(self, resources):
-        """Called by monitor to request resources (EXPERIMENTAL).
+    def request_resources(self, resources: List[dict]):
+        """Called by monitor to request resources.
 
         Args:
-            resources: Either a list of resource bundles or a single resource
-                demand dictionary.
+            resources: A list of resource bundles.
         """
         if resources:
             logger.info(
                 "StandardAutoscaler: resource_requests={}".format(resources))
-        if isinstance(resources, list):
-            self.resource_demand_vector = resources
-        else:
-            # Translate the legacy single resource demand dictionary
-            # to a resource_demand_vector.
-            self.resource_demand_vector = []
-            for resource, count in resources.items():
-                try:
-                    resource_per_worker = self.config["worker_nodes"][
-                        "Resources"][resource]
-                except KeyError:
-                    resource_per_worker = 1  # Assume the worst
-                workers_to_add = int(count / resource_per_worker)
-                if workers_to_add > 0:
-                    self.resource_demand_vector.extend([{
-                        resource: resource_per_worker
-                    }] * workers_to_add)
+        assert isinstance(resources, list), resources
```

Comment:
Can we log this rather than assert it which might break it?

Reply:
This path isn't used

Reply:
It should always be a list; we never call it with anything that's not that.

Reply:
@ericl , this test was basically not a legacy use case or any use case?
https://github.com/ray-project/ray/blob/d1182b827af9fc94be0cd9a6a3211c096aa4c0e3/python/ray/tests/test_autoscaler.py#L629